### ISSUE TITLE
Arrow functions inside of an object getter don't transpile to ES5 properly

### ISSUE THREAD

**Issue by mattlewis92** [Microsoft/TypeScript#15290]
**TypeScript Version:**  2.2.1 / nightly (2.2.0-dev.201xxxxx)
2.2.1

**Code**
You can see the transpiled output [in the playground](https://www.typescriptlang.org/play/#src=const%20example%20%3D%20%7B%0D%0A%20%20arr%3A%20%5B0%2C%201%2C%202%5D%2C%0D%0A%20%20bar()%20%7B%20%7D%2C%0D%0A%20%20get%20foo()%20%7B%0D%0A%20%20%20%20return%20this.arr.map((item)%20%3D%3E%20%7B%0D%0A%20%20%20%20%20%20%2F%2F%20throws%20_this%20is%20undefined%0D%0A%20%20%20%20%20%20return%20this.bar(item)%3B%0D%0A%20%20%20%20%7D)%3B%0D%0A%20%20%7D%0D%0A%7D%3B)
```ts
// A *self-contained* demonstration of the problem follows...
const example = {
  arr: [0, 1, 2],
  bar() { },
  get foo() {
    return this.arr.map((item) => {
      // throws _this is undefined
      return this.bar(item);
    });
  }
};
```

**Expected behavior:**

To transpile to:
```ts
var example = {
    arr: [0, 1, 2],
    bar: function () { },
    get foo() {
        var _this = this;
        return this.arr.map(function (item) {
            return _this.bar(item);
        });
    }
};

```

**Actual behavior:**
Transpiles to:
```ts
var example = {
    arr: [0, 1, 2],
    bar: function () { },
    get foo() {
        return this.arr.map(function (item) {
            // throws _this is undefined
            return _this.bar(item);
        });
    }
};

```

**Comment by sandersn** [comment]
Looks like we just forgot to check for captured this in accessors. Fix is up at #15357

**Comment by mattlewis92** [comment]
Nice one, thanks for the quick turnaround! 🙂
